proposed/http-message.md

```diff
@@ -332,9 +505,11 @@ interface IncomingRequestInterface extends MessageInterface
      * purposes of how duplicate query parameters are handled, and how nested
      * sets are handled.
      *
-     * @return array
+     * @param array $query Array of query string arguments, typically from
+     *     $_GET.
+     * @return void
      */
-    public function getQueryParams();
+    public function setQueryParams(array $query);
```

Comment:
The problem with omitting query parameters from the interface is that we'll get people asking, "How do I get at `$_GET['foo']`?" They want to be able to get at that information without parsing. As such, we need access to it.

While I understand your point of view regarding having query parameters consistent with the URL, I also know that (a) as noted, PHP doesn't do this currently anyways, and (b) most times that you manipulate `$_GET`, you're not concerned about the implications to the URI; you're just stuffing stuff in there such as defaults that you'll want to access later.

Finally, it's a consistency issue: if body and cookie parameters are mutable, query parameters should be as well. At the same time, we're not requiring (and, I argue should not!) that changing these parameters would change the relevant sources (body stream and cookie header, respectively). The point is that the setters are there primarily to allow pre/post processing of the input sources.